tensorflow_addons/__init__.py

```diff
@@ -14,6 +14,47 @@
 # ==============================================================================
 """Useful extra functionality for TensorFlow maintained by SIG-addons."""
 
+
+# Ensure TensorFlow is importable and its version is sufficiently recent. This
+# needs to happen before anything else, since the imports below will try to
+# import tensorflow, too.
+def _ensure_tf_install():
+    """Attempt to import tensorflow, and ensure its version is sufficient.
+    Raises:
+      ImportError: if either tensorflow is not importable or its version is
+      inadequate.
+    """
+    try:
+        import tensorflow as tf
+    except ImportError:
+        # Print more informative error message, then reraise.
+        print(
```

Comment:
I'm not sure this message is needed. A "no module named tensorflow" is already very clear.

If we don't need to print this message, then we don't need to break the flake8 rules and we can also move this check in the "utils" directory to avoid cluttering the init file.

Reply:
Agree that "no module named tensorflow" is sufficient. I'm not sure that we can avoid violating flake8 here though. We'll need to verify the TF version prior to importing all of the TFA submodules so it'll have a import not at top regardless.

PTAL at most recent push

Reply:
True, I'll review when I get home. Thanks for the pull request!